File: python/math/fast_factorize.py
class Fast_factorize:
    def __init__(self, n):
        self.p = [-1]*(n+1)
        
        for i in range(2, n+1):
            if self.p[i]==-1:
                for j in range(i, n+1, i):
                    if self.p[j]==-1:
                        self.p[j] = i
    
    def factorize(self, n):
        res = []
        
        while n>1:
            res.append(self.p[n])
            n //= self.p[n]
        
        return res

File: c++/math/fast_factorize.cpp
struct Fast_factorize {
    vector<int> p;
    
    Fast_factorize(int n) {
        p.resize(n+1);
        fill(p.begin(), p.end(), -1);
        for (int i=2; i<=n; i++) if (p[i]==-1) {
            for (int j=i; j<=n; j+=i) {
                if (p[j]==-1) p[j] = i;
            }
        }
    }
    
    vector<int> factorize(int n) {
        vector<int> res;
        while (n>1) {
            res.pb(p[n]);
            n /= p[n];
        }
        return res;
    }
};
